packages/openapi-ts/src/compiler/typedef.ts

```diff
@@ -123,7 +123,15 @@ export const createTypeRecordNode = (
 ) => {
     const keyNode = createTypeUnionNode(keys);
     const valueNode = createTypeUnionNode(values);
-    const node = ts.factory.createTypeReferenceNode('Record', [keyNode, valueNode]);
+    // NOTE: We use the syntax `{ [key: string]: string }` because using a Record causes
+    //       invalid types with circular dependencies. This is functionally the same.
+    const node = createTypeInterfaceNode([
+        {
+            isRequired: true,
+            name: `[key: ${tsNodeToString(keyNode)}]`,
```

Comment:
shouldn't this always be string?

Reply:
the keyNode is `const keyNode = createTypeUnionNode(keys);`. here we toString it to display it but the key could be a number or `number | string`. tsNodeToString is used due to limitation in the function that I plan to fix later.